openmc/data/data.py

```diff
@@ -313,14 +313,35 @@ def water_density(temperature, pressure=0.1013):
     return coeff / pi / gamma1_pi
 
 
+def zam(name):
+    """Return tuple of (atomic number, mass number, metastable state)
+
+    Parameters
+    ----------
+    name : str
+        Name of nuclide using GND convention, e.g., 'Am242m1'
+
+    Returns
+    -------
+    3-tuple of int
+        Atomic number, mass number, and metastable state
+
+    """
+    symbol, A, state = re.match(r'([A-Zn][a-z]*)(\d+)((?:_[em]\d+)?)',
+                                name).groups()
```

Comment:
Inputting nonsense would yield a relatively cryptic error here (for example ``name = potato`` yields ``AttributeError: 'NoneType' object has no attribute 'groups'``.  Perhaps this can be enclosed in a try-catch block that throws a more friendly error?

Reply:
Good point. @smharper suggested replacing this with my new `openmc.data.zam` function, so I'll add a try/except there.